### DISCUSSION TITLE
An endpoint to update process variables' values

### DISCUSSION THREAD

**Discussion by aarti-km** [elsa-workflows/elsa-core#3866] [Q&A]
Hi,

I am trying to create an API which accepts workflow instance ID, variable name (which is set while designing the workflow by using set variable activity) and new value as parameters, calling on which it should update that particular variable value in the workflow instance which is in suspeneded state.

The variables' vaues are getting updated when I debug but they are not reflecting in the workflow. Is there any way to achieve this? 
(I am using elsa 2).

Thanks

**Comment by sfmskywalker** [comment]
> The variables' vaues are getting updated when I debug but they are not reflecting in the workflow

Do you mean the values are not persisted as part of the workflow instance when you update the variables in the workflow instance and then persist the workflow instance?

**Comment by aarti-km** [comment]
Yes, but I got it resolved. The issue was, I was using the wrong method to save the workflow instance after modifyng the workflow varibales.
Thanks.

**Comment by tsoG2000** [comment]
Hi, @aarti-km and @sfmskywalker I have the same requirement. @aarti-km how did you make that API? 
@sfmskywalker is there some Elsa method which could be used to change the variable of particular workflow instance? 
I'm guessing that directly changing JSON column in Db would be pretty messy. :/

**Reply by sfmskywalker** [reply]
@tsoG2000 Yes, the [Alterations](https://v3.elsaworkflows.io/docs/alterations/introduction) feature was created to allow various alterations to be applied to a given workflow instance, or, indeed as set of workflow instances for bulk processing.

The idea is that there can be various types of alterations, such as "modify variable", which allows you to provide a new value for an existing variable in the workflow instance.

Other alterations are:

- **Migrate**: Migrates a workflow instance to a new version.
- **ScheduleActivity**: Schedules an activity to be executed.
- **CancelActivity**: Cancels an activity. For example, Delay, Event, MessageReceived, etc.

**Reply by tsoG2000** [reply]
@sfmskywalker that is great feature! That is available only on Elsa 3+? I forgot to mention that we're on version 2.9.2.
